From 259e439ad5d20a81d7b437bbcd57003d6fcf7008 Mon Sep 17 00:00:00 2001
From: Philip Muskovac <yofel@gmx.net>
Date: Fri, 21 Jun 2013 01:57:27 +0200
Subject: [PATCH 4/4] fix optional build of akonadi, okular and kate

This reverts the rest of 5fe6f5893b4fb6b2b130ba7f657ff16f9c35862b
looks for smokeakonadi/okular/kate and only builds them if really
everything they need is found.
---
 CMakeLists.txt         |  5 ++---
 modules/CMakeLists.txt | 20 +++++++++++++++-----
 2 files changed, 17 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f6bf35a..45f32fe 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,8 +6,7 @@ find_package(Ruby REQUIRED)
 find_package(KDE4 REQUIRED)
 find_package(QtRuby4 REQUIRED)
 find_package(Smoke REQUIRED 
-    COMPONENTS Akonadi
-        KdeCore
+    COMPONENTS KdeCore
         KdeUi 
         KFile 
         KHtml 
@@ -17,7 +16,6 @@ find_package(Smoke REQUIRED
         KParts
         KTextEditor
         KUtils
-        Okular
         Phonon
         Plasma
         QtCore
@@ -26,6 +24,7 @@ find_package(Smoke REQUIRED
         SopranoClient
         SopranoServer
         Nepomuk )
+find_package(Smoke COMPONENTS Akonadi Okular Kate)
 include(KDE4Defaults)
 include_directories( ${KDE4_INCLUDES} ${SMOKE_INCLUDE_DIR} ${QTRUBY_INCLUDE_DIR} ${SMOKE_QTCORE_INCLUDE_DIR} ${SMOKE_KDECORE_INCLUDE_DIR} )
 
diff --git a/modules/CMakeLists.txt b/modules/CMakeLists.txt
index f494818..f53610f 100644
--- a/modules/CMakeLists.txt
+++ b/modules/CMakeLists.txt
@@ -17,11 +17,11 @@ macro_log_feature(OKULAR_FOUND "Okular" "Okular libraries" "http://www.kde.org"
 
 macro_optional_add_bindings(Soprano_FOUND "Soprano" soprano)
 
-if(KDEPIMLIBS_FOUND AND Akonadi_FOUND)
+if(KDEPIMLIBS_FOUND AND Akonadi_FOUND AND SMOKE_AKONADI_FOUND)
     set(BUILD_Akonadi TRUE)
-else(KDEPIMLIBS_FOUND AND Akonadi_FOUND)
+else(KDEPIMLIBS_FOUND AND Akonadi_FOUND AND SMOKE_AKONADI_FOUND)
     set(BUILD_Akonadi FALSE)
-endif(KDEPIMLIBS_FOUND AND Akonadi_FOUND)
+endif(KDEPIMLIBS_FOUND AND Akonadi_FOUND AND SMOKE_AKONADI_FOUND)
 macro_optional_add_bindings(BUILD_Akonadi "Akonadi" akonadi)
 
 add_subdirectory(kio)
@@ -38,11 +38,21 @@ else(Soprano_FOUND AND Nepomuk_FOUND AND SMOKE_NEPOMUK_FOUND)
 endif(Soprano_FOUND AND Nepomuk_FOUND AND SMOKE_NEPOMUK_FOUND)
 macro_optional_add_bindings(BUILD_Nepomuk "Nepomuk" nepomuk)
 
-macro_optional_add_bindings(OKULAR_FOUND "Okular" okular)
+if(OKULAR_FOUND AND SMOKE_OKULAR_FOUND)
+    set(BUILD_Okular TRUE)
+else(OKULAR_FOUND AND SMOKE_OKULAR_FOUND)
+    set(BUILD_Okular FALSE)
+endif(OKULAR_FOUND AND SMOKE_OKULAR_FOUND)
+macro_optional_add_bindings(BUILD_Okular "Okular" okular)
 
 find_library(KATEINTERFACES_LIBRARY kateinterfaces HINTS ${KDE4_LIB_DIR})
 find_path(KATE_INCLUDE_DIR kate/plugin.h HINTS ${KDE4_INCLUDE_DIR})
 
-macro_optional_add_bindings(KATE_INCLUDE_DIR "Kate" kate)
+if(KATE_INCLUDE_DIR AND SMOKE_KATE_FOUND)
+    set(BUILD_Kate TRUE)
+else(KATE_INCLUDE_DIR AND SMOKE_KATE_FOUND)
+    set(BUILD_Kate FALSE)
+endif(KATE_INCLUDE_DIR AND SMOKE_KATE_FOUND)
+macro_optional_add_bindings(BUILD_Kate "Kate" kate)
 
 macro_display_bindings_log()
-- 
1.8.2.1

